File: app/src/main/java/com/dvinc/circlestimer/di/components/AppComponent.java
/*
 * Copyright (c) 2018 by Denis Verentsov (decsent@Yandex.ru)
 * All rights reserved.
 */

package com.dvinc.circlestimer.di.components;

import com.dvinc.circlestimer.di.modules.AppModule;
import com.dvinc.circlestimer.di.modules.DataModule;
import com.dvinc.circlestimer.di.modules.InteractorsModule;
import com.dvinc.circlestimer.ui.trainings.TrainingsFragment;
import com.dvinc.circlestimer.ui.trainings.newtraining.NewTrainingFragment;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class, DataModule.class, InteractorsModule.class})
public interface AppComponent {
    void inject(TrainingsFragment trainingsFragment);
    void inject(NewTrainingFragment newTrainingFragment);
}

File: app/src/main/kotlin/com/dvinc/workouttimer/presentation/di/component/AppComponent.kt
/*
 * Copyright (c) 2018 by Denis Verentsov (decsent@yandex.ru)
 * All rights reserved.
 */

package com.dvinc.workouttimer.presentation.di.component

import com.dvinc.workouttimer.presentation.di.module.AppModule
import com.dvinc.workouttimer.presentation.di.module.DataModule
import com.dvinc.workouttimer.presentation.di.module.CommonModule
import com.dvinc.workouttimer.presentation.di.module.ViewModelModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [
    AppModule::class,
    DataModule::class,
    CommonModule::class,
    ViewModelModule::class])
interface AppComponent {

    fun getWorkoutComponent(): WorkoutComponent

    fun getExerciseComponent(): ExerciseComponent

    fun getNewWorkoutComponent(): NewWorkoutComponent

    fun getNewExerciseComponent(): NewExerciseComponent
}
